pkg/events/events.go

```diff
@@ -9,24 +9,29 @@ import (
 	k8sv1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1Typed "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
 // Builder provides struct for Event object which contains connection to cluster.
 type Builder struct {
 	// Dynamically discovered Event object.
 	Object *k8sv1.Event
 	// apiClient opens api connection to the cluster.
-	apiClient *clients.Settings
+	apiClient corev1Typed.EventInterface
 	// errorMsg used in discovery function before sending api request to cluster.
 	errorMsg string
 }
 
 // Pull pulls existing Event from cluster.
 func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
+	if apiClient == nil {
+		return nil, fmt.Errorf("apiClient cannot be nil")
```

Comment:
Please add glog before return
glog.V(100).Infof("The apiClient is empty")

Reply:
Done.